ext/dom/xpath_rust.rs

```diff
@@ -0,0 +1,56 @@
+use std::ffi::{CString, c_char};
+
+
+#[no_mangle]
+pub extern "C" fn domxpath_quote_literal(input: *const c_char, len: *mut usize) -> *mut c_char {
+    let slice = unsafe { std::slice::from_raw_parts(input as *const u8, *len as usize) };
```

Comment:
This should preferably contain a "safety requirement" comment, i.e. that it is expected that the caller does not hold any mutable references to the input. I know it's a bit trivial in this case, but when using unsafe I'd rather see it documented, and the caller must know about it too to avoid soundness issues.